scripts/install_avalanchego_release.sh

```diff
@@ -11,41 +11,119 @@ source "$SUBNET_EVM_PATH"/scripts/versions.sh
 # Load the constants
 source "$SUBNET_EVM_PATH"/scripts/constants.sh
 
-VERSION=$AVALANCHEGO_VERSION
-
 ############################
 # download avalanchego
 # https://github.com/ava-labs/avalanchego/releases
 GOARCH=$(go env GOARCH)
 GOOS=$(go env GOOS)
-BASEDIR=${BASE_DIR:-"/tmp/avalanchego-release"}
+BASEDIR=${BASEDIR:-"/tmp/avalanchego-release"}
+
 mkdir -p ${BASEDIR}
+
+VERSION=$AVALANCHEGO_VERSION
+
 AVAGO_DOWNLOAD_URL=https://github.com/ava-labs/avalanchego/releases/download/${VERSION}/avalanchego-linux-${GOARCH}-${VERSION}.tar.gz
 AVAGO_DOWNLOAD_PATH=${BASEDIR}/avalanchego-linux-${GOARCH}-${VERSION}.tar.gz
+
 if [[ ${GOOS} == "darwin" ]]; then
   AVAGO_DOWNLOAD_URL=https://github.com/ava-labs/avalanchego/releases/download/${VERSION}/avalanchego-macos-${VERSION}.zip
   AVAGO_DOWNLOAD_PATH=${BASEDIR}/avalanchego-macos-${VERSION}.zip
 fi
 
-AVALANCHEGO_BUILD_PATH=${AVALANCHEGO_BUILD_PATH:-${BASEDIR}/avalanchego-${VERSION}}
-mkdir -p $AVALANCHEGO_BUILD_PATH
+BUILD_DIR=${AVALANCHEGO_BUILD_PATH}-${AVALANCHEGO_VERSION}
```

Comment:
This seems to fail locally unless you specify `AVALANCHEGO_BUILD_PATH` (this passes CI since it manually populated this env var)

If we are going to use it in `BUILD_DIR` here, then we should update this to ensure that if it's not populated it defaults to a reasonable location like the previous behavior defaulting to use the already populated `BASEDIR`

Reply:
Agreed! Will change